cmd/kubeadm/app/util/endpoint.go

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"net"
+)
+
+// GenerateMasterEndpoint generates a properly formatted Master URL.
+func GenerateMasterEndpoint(address string, port int32) string {
+	var masterEndpoint string
```

Comment:
If the func does not perform any validation, then we might as well dump it and simply call the net.JoinHostPort func. I'm going to update the util func to use JoinHostPort and perform addr validation.